src/condor_scripts/condor_upgrade_check

```diff
@@ -550,6 +550,179 @@ def warn_user_records():
+        has_cpu_busy_time = []
+
+        p = subprocess.run(["condor_config_val", "-dump"], stdout=subprocess.PIPE)
+        cmd_output = p.stdout.rstrip().decode()
+        has_problem = False
+        for line in cmd_output.split("\n"):
+            line = line.strip()
+            check = line.lower()
+            parts = check.split("=", 1)
+            if len(parts) < 2:
+                continue
+            for attr in attrs:
+                if attr.lower() in parts[1]:
+                    if attr == "CpuBusy":
+                        pos = check.find("cpubusy")
+                        false_positive = True
+                        while pos != -1:
+                            if check.find("cpubusytime", pos) != pos:
+                                false_positive = False
+                                break
+                            pos = check.find("cpubusy", pos + 1)
+                        if false_positive:
+                            continue
+                    not_in_other_def = True
+                    for other in attrs:
+                        if other.lower() == parts[0].strip():
+                            not_in_other_def = False
+                            break
+                    if not_in_other_def:
+                        found_attrs.append(attr)
+            if "cpubusytime" in check and line != default_suspend_line:
+                has_cpu_busy_time.append(line)
+                has_problem = True
+
+        knobs = [
+            "IS_OWNER",
+            "KILL",
+            "PREEMPT",
+            "SLOTS_CONNECTED_TO_KEYBOARD",
+            "SUSPEND",
+        ]
+        count_using_policy = 0
+        for knob in knobs:
+            p = subprocess.run(["condor_config_val", "-v", knob], stdout=subprocess.PIPE)
+            cmd_output = p.stdout.rstrip().decode()
+            for line in cmd_output.split("\n"):
+                line = line.upper()
+                if "POLICY:DESKTOP" in line or "POLICY:UWCS_DESKTOP" in line:
+                    count_using_policy += 1
+        if count_using_policy >= 3:
+            found_attrs.clear()
+
+        custom_set = []
+        for attr in found_attrs:
+            p = subprocess.run(["condor_config_val", "-v", attr], stdout=subprocess.PIPE)
+            cmd_output = p.stdout.rstrip().decode()
+            is_default = False
+            for line in cmd_output.split("\n"):
+                if "<default>" in line.lower():
+                    has_problem = True
+                    is_default = True
+                    break
+            if not is_default:
+                custom_set.append(attr)
+
+        for attr in custom_set:
+            found_attrs.remove(attr)
+
+        if has_problem:
+            format_print(
+                "Error checking for desktop policy configuration macros:", newline=True
+            )
+            need_newline = False
+            if len(found_attrs) > 0:
+                found_attrs.sort()
```

Comment:
`sorted()` will convert a set to a list and sort it.
```suggestion
                found_attrs = sorted(found_attrs)
```
Consider batching this with the previous suggestion.